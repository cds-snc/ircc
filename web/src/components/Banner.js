import React from 'react'
import styled from 'styled-components'
import { GoCSignature } from '@cdssnc/gcui'
import PropTypes from 'prop-types'

const Padded = styled.div`padding: 0.8em 0.9em;`

const AlignedRight = styled.div`text-align: right;`

const BannerBody = styled.div`
  margin-right: auto;
  margin-left: auto;
  padding: 15px 1em;
  @media (min-width: 1200px) {
    width: 61%;
  }
  @media (min-width: 992px) {
    width: 970px;
  }
  @media (min-width: 768px) {
    width: 750px;
  }
  &:after,
  &:before {
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
  }
`

const LanguageSwitcher = styled.a`
  text-decoration: underline;
  color: #284162;
  background-color: none;
  &:visited {
    color: #7834bc;
  }
  &:hover {
    cursor: pointer;
  }
`

const Banner = ({ lang, changeLanguage }) => (
  <BannerBody>
    <AlignedRight>
      <LanguageSwitcher onClick={changeLanguage}>
        {lang === 'en' ? 'Fran√ßais' : 'English'}
      </LanguageSwitcher>
    </AlignedRight>
    <Padded>
      <GoCSignature width="26em" />
    </Padded>
  </BannerBody>
)

Banner.propTypes = {
  lang: PropTypes.string.isRequired,
  changeLanguage: PropTypes.func.isRequired,
}

export default Banner
