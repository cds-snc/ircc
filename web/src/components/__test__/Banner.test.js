import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import Banner from '../Banner'

describe('<Banner />', () => {
  it('can be rendered', () => {
    let wrapper = shallow(<Banner lang="en" changeLanguage={jest.fn()} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
