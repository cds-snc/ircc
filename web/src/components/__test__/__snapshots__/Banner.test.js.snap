// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> can be rendered 1`] = `
<Styled(div)>
  <Styled(div)>
    <Styled(a)
      onClick={[Function]}
    >
      Fran√ßais
    </Styled(a)>
  </Styled(div)>
  <Styled(div)>
    <Wordmark
      width="26em"
    />
  </Styled(div)>
</Styled(div)>
`;
